doc/service.md

```diff
@@ -0,0 +1,85 @@
+What is a service?
+==================
+
+A service for Cozy is defined by these criteria:
+
+1. It's some JavaScript code
+2. That is embedded in Cozy-Proxy
+3. As express/americano apps
+4. And exposes a REST API in JSON
+5. to offer functionalities to cozy applications.
+6. This API follows the [JSON api specification](http://jsonapi.org/)
+7. And [Cozy guidelines](https://cozy.github.io/cozy-guidelines/)
+8. With URL prefixed by `/services/:service-name`.
+9. It benefits from Cozy authentication and permissions
+10. And workflows (code review, i18n).
+
+
+Rationale
+---------
+
+### 1. It's some JavaScript code
+
+JS is the lowest common denominator for all the code written at Cozy.
+Technically, a service can be written in something else (CoffeeScript or
+TypeScript) but it has to be compiled simply to JavaScript.
+
+### 2. That is embedded in Cozy-Proxy
+
+I can think to several services than can be useful: files, notifications,
+applications management, sharing, settings, indexer/search, etc.
+Running one node processus for each service will take a lot of RAM.
+[Simple Cozy](https://github.com/cozy/simple-cozy) has shawn that it's
+possible to run a nodejs processus with code from several git repository to
+use less memory.
+
+I think to embed the services in cozy-proxy and not cozy-data-system beacuse
+cozy-data-system could become a service its-self (some day, in the long term).
+And most server-side code from cozy-home too. Maybe, we can even unify all of
+our nodejs code to run in the same processus (I'm dreaming).
+
+### 3. As express/americano apps
+
+It's the simple way I see to make this happen.
+
+### 4. And exposes a REST API in JSON
+
+Idem. I don't see a reason to choose something more complicated.
+
+### 5. to offer functionalities to cozy applications.
+
+Cozy applications can be run in the browser with cozysdk, on the server,
+but there are also cozy-mobile and cozy-desktop. All of them should be able to
+use the services even if the authentication mechanisms are not the same.
+
+### 6. This API follows the [JSON api specification](http://jsonapi.org/)
+
+I have nothing against [JSON-ld](http://json-ld.org/). The important is to
+take one standard and stick to it. JSON api looks good. So, go for it!
+
```

Comment:
PSON is no longer maintained. ProtocolBuffer enforces using a schema. Both are not as ubiquitous as JSON (for example, JSON is displayed nicely in chrome and firefox devtools, you can read it when using curl, you can use it with almost any programming language). And for performances, it's not clear if the download size (> 10ko) is not larger than the bytes transfered on calls. Maybe, JSON with brotli compression is even smaller than ProtocolBuffer.